templates/commands/render/action.go

```diff
@@ -141,6 +141,29 @@ func templateAndCompileRegexes(regexes []model.String, scope *common.Scope) ([]*
 	return compiled, merr
 }
 
+// processPaths processes a list of input String paths for go templating, relative paths,
+// OS-specific slashes, and (soon) file globbing.
+func processPaths(paths []model.String, scope *common.Scope) ([]model.String, error) {
+	pathsCopy := make([]model.String, len(paths))
+	copy(pathsCopy, paths)
+
+	var err error
+	for i, p := range pathsCopy {
+		p.Val, err = parseAndExecuteGoTmpl(p.Pos, p.Val, scope)
```

Comment:
The calls to `p.Val = ...` are only safe because the input is not a pointer (`[]model.String` vs `[]*model.String`). If that ever changes to a pointer, this would be modifying the result in place.

It might be safer and more idiomatic to either:

1. Accept the slice of pointers, deep-copy each element, return a new slice
2. Accept the slice of pointers, modify in place, return only an error (because the provided slice will have been modified).

I'm just a little worried about the probability of a future bug here given the implicit dependency on copy-by-value.

Non-blocking.

Reply:
Changed to append to empty slice without modifying in place.